### ISSUE TITLE
Crash: collection was mutated while being enumerated.

### ISSUE THREAD

**Issue by kevinrenskers** [Instabug/Instabug-iOS#229]
```
*** Terminating app due to uncaught exception 'NSGenericException', reason: '*** Collection <__NSArrayM: 0x1c0244440> was mutated while being enumerated.'

Last Exception Backtrace:
0   CoreFoundation                      0x0000000185aae364 <redacted> + 124
1   libobjc.A.dylib                     0x0000000184cf4528 objc_exception_throw + 56
2   CoreFoundation                      0x0000000185aad9f0 <redacted> + 0
3   InstabugCore                        0x0000000103ed8e38 -[IBGCache sendAll] + 192
4   InstabugCore                        0x0000000103e9a9f0 -[IBGCacheManager send] + 216
5   InstabugCore                        0x0000000103e9a510 __42-[IBGCacheManager initializationOperation]_block_invoke + 36
6   InstabugCore                        0x0000000103e9f8e4 -[IBGOperation start] + 168
7   Foundation                          0x0000000186487894 <redacted> + 404
8   libdispatch.dylib                   0x000000018542aa14 <redacted> + 16
9   libdispatch.dylib                   0x0000000185432f08 <redacted> + 428
10  libdispatch.dylib                   0x000000018543180c <redacted> + 604
11  libdispatch.dylib                   0x0000000185437cf4 <redacted> + 600
12  libdispatch.dylib                   0x0000000185437a38 <redacted> + 120
13  libsystem_pthread.dylib             0x00000001856d306c _pthread_wqthread + 1268
14  libsystem_pthread.dylib             0x00000001856d2b6c start_wqthread + 4
```

Crash in 7.4. Again, don't know if it was fixed in later versions but haven't seen this one in the github issues.

**Comment by ZiadAyman** [comment]
This one is fixed, it has a PR that is ready for testing. 
Will be included in our coming release very soon :) 

I will let you know once the fix is released.

**Comment by ZiadAyman** [comment]
Hey @kevinrenskers 
This is fixed and released on iOS SDK version 7.8.2.  

I will close this for now, give it a try and if anything is not in place make sure to re-open the issue and notify me!
